File: coin_change_2.py
def change(amount,coins):
	table=[[0]*(amount+1) for i in range(len(coins)+1)]
	
	for i in range(len(table)):
		table[i][0]=1
	
	
	n=len(coins)+1
	m=amount+1
	
	for row in range(1,n):
		for col in range(1,m):
			if col-coins[row-1]>=0:
				table[row][col]=table[row][col-coins[row-1]]+table[row-1][col]
			else:
				table[row][col]=table[row-1][col]
				
	return table[-1][-1]



print(change(5,[1,2,5]))
	

File: coin_change_2.c
#include<stdio.h>
int change(int amount, int* coins, int n){

    
	if(amount=0){
		printf("here");
		return 1;
	}
	if(amount<0||n==0){
		return 0;
	}
	int ans=0;
	//consider inclusion or removal of last coin
	int include=change(amount-coins[n],coins,n);
	int exclude=change(amount,coins,n-1);
	
	
	printf("%d",include+exclude);
	return include+exclude;
    
}
int main()
{
	
	int S[] = {1,2,5};
	int n = 3;
	int N = 5;
	int ans=change(N,S,n);
	printf("hi: %d",ans);
	
}
